### PR TITLE
Merging after dividing IRequests and creation interface to AnswerContainer class

### PR DESCRIPTION
**Pull request by PaulMoroz**
## Code reviewers

- [ ] @IhorShevchuk @Yurazast @Vitalik-Kshly @losimen @AndrianKupyniak 

### Second Level Review

- [ ] @github_username

## Summary of issue

I was need to divide IRequests into small files and create interface to answercontainer class

## Summary of change

I divided to divide IRequests into small files and created interface to answercontainer class

### CODE REVIEW

File: ThreadWorker.h [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
-public:
-    ThreadWorker();
-    ~ThreadWorker();
-    std::vector<ThreadInfo> m_ThreadPool;
-    void PushRequest(IRequests req);
-
-};
-
-#endif //BACKEND_THREADWORKER_H
+//
```
Comment (IhorShevchuk) [inline ThreadWorker.h:1]: I do not see changes in this file, how is it possible that it got into pull request?

File: IRequests/RequestSignUp.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include "IRequests.h"
+#include "RequestSignUp.h"
+#include <cpprest/json.h>
+#include "../stringtowstring.h"
+using namespace web;
+
+RequestSignUp::RequestSignUp(MSSQLDatabase* d,answercontainerinterface* answercontainer, const ISXModel::User& user) : IRequests(db, answercontainer), user(user) {}
```
Comment (IhorShevchuk) [inline IRequests/RequestSignUp.cpp:8]: you are passing 'answercontainerinterface' here as parameter but I do not see usage of it in this class.

File: AnswerContainer.h [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
-#include "irequests.h"
+#include "IRequests/IRequests.h"
 #include "MSSQLDatabase.h"
+#include "AnswerContainerInterface.h"
 using namespace web;
 using namespace web::http;
 using namespace web::http::experimental::listener;
 
-class answercontainer {
+class answercontainer: public answercontainerinterface{
```
Comment (IhorShevchuk) [inline AnswerContainer.h:11]: Name convention for those to classes are not followed. Please rename them.

Comment (Yurazast) [inline AnswerContainer.h:11]: It seems to me that it is better to call it IAnswerContainer instead of answercontainerinterface.

**Pull request closed without merging**